### PR TITLE
Add promise polyfill in order to support IE11

### PR DESCRIPTION
**Pull request by mamont**

### CODE REVIEW

**Comment by markandrus** [comment]
Thanks, @mamont. Could you please add the contents of [es6-promise's LICENSE](https://github.com/stefanpenner/es6-promise/blob/master/LICENSE) to our [LICENSE.md](https://github.com/twilio/twilio-common.js/blob/master/LICENSE.md)?

File: lib/accessmanager.js [hunk 1]
```diff
@@ -3,6 +3,8 @@
 var EventEmitter = require('events').EventEmitter;
 var inherits = require('util').inherits;
```
Comment (markandrus) [inline lib/accessmanager.js]: Can you delete this empty line?

File: lib/accessmanager.js [hunk 2]
```diff
@@ -3,6 +3,8 @@
...
 
+var Promise = require('es6-promise').Promise;
```
Comment (markandrus) [inline lib/accessmanager.js]: It appears that we would always use the polyfill if we did it this way. Seems like we should instead do something like

``` js
var Promise = typeof Promise === 'undefined' ? require('es6-promise').Promise : Promise;
```

I also wonder if this form will pass linting.

**Comment by liberty-rowland** [comment]
Should we really roll a poly-fill into all of our builds? This only affects IE11, no need to include this for other browsers. Additionally, we're going to need to duplicate this poly-fill in every product we want to support IE11. Can we just recommend a poly-fill for IE11 support?

**Comment by mamont** [comment]
IE11 is one of very common browsers for our customers. It would be weird to force them to make hacks around our library to support it.
I'm ok with doing nothing on our side to support some marginal browsers and versions. But not for one from the mainstream.
So please let's find some way to add IE11 support.

**Comment by markandrus** [comment]
+1, thanks @mamont

**Pull request merged**